src/main/java/org/sagebionetworks/web/client/widget/ReactComponent.java

```diff
@@ -143,15 +143,22 @@ private void injectChildWidgetsIntoComponent() {
 
   public void render(ReactElement<?> reactElement) {
     this.reactElement = reactElement;
-    maybeCreateRoot();
     injectChildWidgetsIntoComponent();
 
     // This component may be a React child of another component. If so, we must rerender the ancestor component(s) so
     // that they use the new ReactElement created in this render step.
     ReactComponent<?> componentToRender = getRootReactComponentWidget();
     if (componentToRender == this) {
-      // Resynchronize with the DOM
-      root.render(this.reactElement);
+      // Asynchronously schedule creating a root in case React is still rendering and may unmount the current root
+      Timer t = new Timer() {
+        @Override
+        public void run() {
+          maybeCreateRoot();
+          // Resynchronize with the DOM
+          root.render(reactElement);
+        }
+      };
+      t.schedule(0);
```

Comment:
Follow up from #5497, the call to `render` also needs to be scheduled or else the old root might still exist.

Reply:
Fixes an e2e test